packages/block-library/src/template-part/edit/selection/template-part-previews.js

```diff
@@ -111,23 +111,37 @@ function TemplatePartsByArea( {
 	area = 'uncategorized',
 	labelsByArea,
 } ) {
+	const templatePartsToShow =
+		templateParts.filter(
+			( templatePart ) =>
+				'uncategorized' === area || templatePart.area === area
+		) || [];
```

Comment:
Should we `memo` this? In case of a huge amount of template parts it might get slow.

Reply:
100%, this can be moved into the same memo noted above regarding the return suggestion.